### PR TITLE
[ENH] Update lcss distance

### PR DESCRIPTION
**Pull request by chrisholder**
#### Reference Issues/PRs
This pr depends on pr #323 

#### What does this implement/fix? Explain your changes.
This pr updates lcss to the new format for distances. In addition in doing so it makes optimisations for computing the distances between multiple time series. 

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.
- [x] For new estimators, I've added the estimator to the online documentation.

### CODE REVIEW

File: aeon/distances/_alignment_paths.py [hunk 1]
```diff
@@ -84,24 +84,25 @@ def compute_lcss_return_path(
...
+    y_size = y.shape[1]
 
-    i, j = (x_size, y_size)
+    i, j = (x_size - 1, y_size - 1)
     path = []
 
     while i > 0 and j > 0:
         if bounding_matrix[i - 1, j - 1]:
-            if _univariate_squared_distance(x[:, i - 1], y[:, j - 1]) <= epsilon:
+            if _univariate_euclidean_distance(x[:, i - 1], y[:, j - 1]) <= epsilon:
```
Comment (GuzalBulatova) [inline aeon/distances/_alignment_paths.py:95]: I don't entirely understand the difference here, but just to double-check: above the i and j already got subtracted 1, here we do it again, is that right? 
For my benefit: for example for 1D x and y inputs the cost_matrix.shape == (len(x), len(y))?

**Review by GuzalBulatova — COMMENTED** [review]
Amazing contribution, @chrisholder! A particular appreciation for descriptive docstrings 🚀
I haven't gone through all the changes yet though, have a couple of questions so far.

File: aeon/distances/_alignment_paths.py [hunk 2]
```diff
@@ -84,24 +84,25 @@ def compute_lcss_return_path(
...
                 path.append((i - 1, j - 1))
                 i, j = (i - 1, j - 1)
-            elif cost_matrix[i - 1][j] > cost_matrix[i][j - 1]:
+            elif cost_matrix[i - 1][j] > cost_matrix[i - 1][j]:
```
[the remark this answers is not in the document]
Comment (chrisholder) [reply aeon/distances/_alignment_paths.py:98]: This is a great spot! You are correct just a typo I will fix it now!

File: aeon/distances/_alignment_paths.py (hunk 1 above)
Comment (chrisholder) [reply aeon/distances/_alignment_paths.py:95]: So the cost matrix that lcss returns is 1 bigger than x.shape[1]. So a cost matrix of two 1D univariate timeseries of length 10 would be 11x11. As such we look at the previous value going back hence the -1.

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**